src/main/java/model/position/Position.java

```diff
@@ -0,0 +1,71 @@
+package model.position;
+
+import java.util.HashSet;
+import java.util.Objects;
+import java.util.Set;
+import model.direction.MovingPattern;
+
+public class Position {
+    private final int file;
+    private final int rank;
+
+    private Position(int file, int rank) {
+        this.file = file;
+        this.rank = rank;
+    }
+
+    private static final Set<Position> cache = new HashSet<>();
+
+    static {
+        for (int file = 1; file <= 8; file++) {
+            initRank(file);
+        }
+    }
+
+    private static void initRank(int file) {
+        for (int rank = 1; rank <= 8; rank++) {
+            cache.add(new Position(file, rank));
+        }
+    }
+
+    public boolean isAvailablePosition(MovingPattern movingPattern) {
+        int movedFile = file + movingPattern.fileDifferential();
+        int movedRank = rank + movingPattern.rankDifferential();
+        return 1 <= movedFile && movedFile <= 8 && 1 <= movedRank && movedRank <= 8;
```

Comment:
File, Rank로 자료형을 다루었다면 가로세로 범위가 1 ~ 8 이라는 불필요한 정보를 Position이 기억하지 않아도 될거같아요.
```suggestion
// 메서드명은 대충 지음
        return file.available(movingPattern.fileDifferential()) && rank.available(movingPattern.rankDifferential());
```

Reply:
ㅎ 다른 크루들한테서 항상 File과 Rank 객체를 보다보니 착각하고 직전 리뷰를 적긴 했네요!
그래도 file과 rank가 원시값으로 다뤄지고 있으니 랩핑해보는건 어떨까요?

Reply:
`File`과 `Rank` 모두 랩핑하였습니다!
해당 객체로 랩핑하는 과정에서 현재 `File`과 `Rank` 모두 사용자의 입력과 매칭될 수 있는 `expression` 필드를 가지고 있습니다.
이 때 해당 부분은 도메인(`File`, `Rank`)이 뷰 로직을 알고있다고 생각되지 않고, 뷰에 의존적이지 않다고 생각되어 구성하게 되었습니다!
혹시 생각이 잘못된 부분이 있으면 말씀 부탁드리겠습니다!

Reply:
요구사항에 맞춘 구현이라는 생각은 드는데요, (최초에 consoleView에서 입력받으려면 a1 등으로 구현되는 게 유리함)
이정도는 저도 도메인 로직이라는 생각이 듭니다 ㅎ (실제 체스에서 a1 등이 활용되므로)